### PR TITLE
sources: add spnego

### PR DESCRIPTION
**Pull request by rissson**
## Details

Add a SPNEGO source. Allows login using an existing Kerberos identity.

TODO:
- [x] update CI & docker image to add required system libraries for GSSAPI
- [x] update CI to install required tools to run SPNEGO tests
- [x] documentation

---

## Checklist

-   [x] Local tests pass (`ak test authentik/`)
-   [x] The code has been formatted (`make lint-fix`)

If an API change has been made

-   [x] The API schema has been updated (`make gen-build`)

If changes to the frontend have been made

-   [ ] The code has been formatted (`make web`)
-   [x] The translation files have been updated (`make i18n-extract`)

If applicable

-   [x] The documentation has been updated
-   [x] The documentation has been formatted (`make website`)

cc. @Mareo

### CODE REVIEW

File: authentik/sources/spnego/views.py [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+            "if/error.html",
+            context={
+                "title": _("SPNEGO authentication required"),
+                "message": _("""
+                    Make sure you have valid tickets (obtainable via kinit) and configured the browser correctly.
+                    You can find out more at https://goauthentik.io/integrations/sources/spnego/browser
+                """),
+            },
+            status=401,
+        )
```
Comment (rissson) [inline authentik/sources/spnego/views.py:46]: We may want to handle this differently.

**Review by BeryJu — COMMENTED** [review]
I don't think I know enough about kerberos to review this lol

**Comment by rissson** [comment]
Actually I have an even better implementation in progress that supports user sync and password writeback. It'll be an ok solution to use until we have a proper kerberos provider setup.

File: authentik/sources/spnego/views.py [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+            out_token = server_ctx.step(in_token)
+        except gssapi.exceptions.GSSError as exc:
+            LOGGER.debug("GSSAPI security context failure", exc=exc)
+            return self.challenge(request)
+
+        if not server_ctx.complete or server_ctx.initiator_name is None:
+            self.set_server_ctx(state, server_ctx)
+            return self.challenge(request, out_token)
+
+        identifier = str(server_ctx.initiator_name)
```
Comment (rissson) [inline authentik/sources/spnego/views.py:133]: This is case insensitive. Better make sure it's always stored lower/uppercase

File: authentik/sources/spnego/views.py [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+class SPNEGOView(View):
+    """SPNEGO login"""
+
+    source: SPNEGOSource
+
+    def challenge(self, request, token: str | None = None) -> HttpResponse:
+        """Get SNPEGO challenge response"""
+        response = render(
+            request,
+            "if/error.html",
```
Comment (BeryJu) [inline authentik/sources/spnego/views.py:37]: TODO: see if we can do this without rendering a full static html page

**Comment by rissson** [comment]
Closing in favor of https://github.com/goauthentik/authentik/pull/10815

**Pull request closed without merging**